pingcap/tidb-operator
# Improve `toolImage` to be able to set without image tag

### What problem does this PR solve?

Closes #4047 
### What is changed and how does it work?


### Code changes


### Tests



The TiKV's version is v5.0.1
* Deploy `Backup` with `spec.toolImage: `, then check if  image of Pod is 'pingcap/br:v5.0.1'.
* Deploy `Backup` with `spec.toolImage: pingcap/br:v5.0.2`, then check if  image of Pod is 'pingcap/br:v5.0.2'.
* Deploy `Backup` with `spec.toolImage: myimage`, then check if  image of Pod is 'myimage: v5.0.1'.

Jobs of all `Backup` is completed.
* Deploy `Restore` with `spec.toolImage: `, then check if  image of Pod is 'pingcap/br:v5.0.1'.
* Deploy `Restore` with `spec.toolImage: pingcap/br:v5.0.2`, then check if  image of Pod is 'pingcap/br:v5.0.2'.
* Deploy `Restore` with `spec.toolImage: myimage`, then check if  image of Pod is 'myimage: v5.0.1'.

Jobs of all `Restore` is completed.

### Side effects


### Related changes


### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Use tool image of the `TiKV`'s version if tool image of `BR` does not contains tag
```
